subcommands/targets/offline-update.go

```diff
@@ -72,18 +80,42 @@ func doOfflineUpdate(cmd *cobra.Command, args []string) {
 	targetName := args[0]
 	dstDir := args[1]
 
-	if len(ouTag) == 0 {
-		subcommands.DieNotNil(errors.New("missing mandatory flag `--tag`"))
+	if len(ouTag) == 0 && len(ouWave) == 0 {
+		subcommands.DieNotNil(errors.New("one of the mandatory parameters `--tag` or `--wave` should be specified"))
+	}
+	if len(ouTag) > 0 && len(ouWave) > 0 {
+		subcommands.DieNotNil(errors.New("either `--tag` or `--wave` should be specified exclusively"))
+	}
+	if len(ouWave) > 0 && ouProd {
+		subcommands.DieNotNil(errors.New("`the --prod` flag is redundant if the `--wave` parameter is specified"))
 	}
 
-	fmt.Printf("Checking whether Target exists; target: %s, tag: %s, production: %v\n", targetName, ouTag, ouProd)
-	subcommands.DieNotNil(checkIfTargetExists(factory, targetName, ouTag, ouProd))
-
-	ti, err := getTargetInfo(factory, targetName)
-	subcommands.DieNotNil(err, "Failed to obtain Target's details:")
+	var targetCustomData *tuf.FileMeta
+	var targetGetErr error
+	// Get the wave/prod/CI specific target with the specified tag to check if it is really present
+	if len(ouWave) > 0 {
+		fmt.Printf("Getting Wave Target details; target: %s, wave: %s...\n", targetName, ouWave)
+		_, targetGetErr = getWaveTargetMeta(factory, targetName, ouWave)
+	} else if ouProd {
+		fmt.Printf("Getting production Target details; target: %s, tag: %s...\n", targetName, ouTag)
+		_, targetGetErr = getProdTargetMeta(factory, targetName, ouTag)
+	} else {
+		fmt.Printf("Getting CI Target details; target: %s, tag: %s...\n", targetName, ouTag)
+		_, targetGetErr = getCiTargetMeta(factory, targetName, ouTag)
+	}
+	subcommands.DieNotNil(targetGetErr)
+	// Get the specified target from the list of factory targets to obtain the "original" tag/branch that produced
+	// the target, so we can find out the correct app bundle fetch URL.
+	// TBD: remove it once we do something so the wave/prod/CI specific target include enough info to deduce
+	// the download URLs
```

Comment:
What you put in a (TBD) comment is our _future roadmap_; so let it stay in a roadmap.
Let's keep the code clean, and not create a confusion as if this feature is not fully implemented.
Later we will simply extend this feature with an improved/better way... a long-lasting TBD does not help with that.

Reply:
Agree, I put the comment basically for the reviewers, it might be useful to understand why I called the methods `get*TargetMeta` and why it returns a value.

Reply:
removed the TBD message dfb80e8.